crates/rpc/rpc/src/debug.rs

```diff
@@ -550,6 +553,90 @@ where
             .await
     }
 
+    /// The `debug_executionWitness` method allows for re-execution of a block with the purpose of
+    /// generating an execution witness. The witness comprises of a map of all hashed trie nodes
+    /// to their preimages that were required during the execution of the block, including during
+    /// state root recomputation.
+    pub async fn debug_execution_witness(
+        &self,
+        block: BlockNumberOrTag,
+    ) -> Result<HashMap<B256, Bytes>, Eth::Error> {
+        let block = match self.inner.eth_api.block(block.into()).await? {
+            None => return Err(EthApiError::UnknownBlockNumber.into()),
+            Some(res) => res,
+        };
+        let (cfg, block_env, _) = self.inner.eth_api.evm_env_at(block.hash().into()).await?;
```

Comment:
this can be done concurrently

ref 

https://github.com/paradigmxyz/reth/blob/a28b2f09da658e7c7bc1c6456f70293135ad4904/crates/rpc/rpc/src/debug.rs#L196-L199

Reply:
good point